tests/components/tradfri/test_config_flow.py

```diff
@@ -218,16 +221,23 @@ async def test_import_connection_legacy_no_groups(
 
 
 async def test_discovery_duplicate_aborted(hass):
-    """Test a duplicate discovery host is ignored."""
-    MockConfigEntry(domain="tradfri", data={"host": "some-host"}).add_to_hass(hass)
+    """Test a duplicate discovery host aborts and updates existing entry."""
+    entry = MockConfigEntry(
+        domain="tradfri", data={"host": "some-host"}, unique_id="homekit-id"
+    )
+    entry.add_to_hass(hass)
 
     flow = await hass.config_entries.flow.async_init(
-        "tradfri", context={"source": "zeroconf"}, data={"host": "some-host"}
+        "tradfri",
+        context={"source": "homekit"},
+        data={"host": "new-host", "properties": {"id": "homekit-id"}},
     )
 
     assert flow["type"] == data_entry_flow.RESULT_TYPE_ABORT
     assert flow["reason"] == "already_configured"
 
+    assert entry.data["host"] == "new-host"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5422434)